src/js/tls.js

```diff
@@ -68,11 +70,23 @@ TLSSocket.prototype.write = function(data, callback) {
   if (!Buffer.isBuffer(data)) {
     data = new Buffer(data);
   }
-
-  data = this._write(data);
-  return this._socket.write(data, callback);
+  if (!this.secureConnectionDone) {
+    if (this.buff.length > 0) {
+      this.buff = Buffer.concat([this.buff, data]);
+      return;
+    }
+    this.buff = data;
+    return;
+  } else {
+    return writeOut(this, data, callback);
+  }
```

Comment:
Looks like a manual implementation. Does writable provide any interface for this? If yes, perhaps we could inherit it.